src/components/sbb-dialog/sbb-dialog.tsx

```diff
@@ -254,12 +256,21 @@ export class SbbDialog implements ComponentInterface {
 
   // Check if the pointerdown event target is triggered on the dialog.
   private _pointerDownListener = (event: PointerEvent): void => {
-    this._isPointerDownEventOnDialog = isEventOnElement(this._dialog, event);
+    this._isPointerDownEventOnDialog = event
+      .composedPath()
+      .filter((e) => e instanceof window.HTMLElement)
+      .some((target) => (target as HTMLElement).tagName === this._dialogId);
   };
 
   // Close dialog on backdrop click.
   private _closeOnBackdropClick = async (event: PointerEvent): Promise<void> => {
-    if (!this._isPointerDownEventOnDialog && !isEventOnElement(this._dialog, event)) {
+    if (
+      !this._isPointerDownEventOnDialog &&
+      !event
+        .composedPath()
+        .filter((e) => e instanceof window.HTMLElement)
```

Comment:
Nitpick: You could use typescript assertion to not have to cast to HTMLElement in some
```suggestion
        .filter((e): e is HTMLElement => e instanceof window.HTMLElement)
```